T2-TECHNICAL-OVERVIEW.md

```diff
@@ -0,0 +1,137 @@
+
+The CLI is the primary method of interacting with Tessel for users and develoeprs. It provides useful functions like listing available Tessels, deploying code, and setting wifi credentials on the device.
+
+#####[Tessel 2 Virtual Machine](https://www.github.com/tessel/t2-vm) (VM)
+The VM is used primarily by Tessel developers who either don't have hardware available or want to develop on a faster computer. The repo provides the ability to create and run these Tessel virtual machines. You can use the CLI (see above) to interact with the running VM just as you would an actual Tessel 2. The VM has the limtation of not being able to set wifi credentials (it passes through the host network interface) or using hardware modules but USB devices do get passed through.
+
+#####[Tessel 2 Compiler](https://www.github.com/tessel/t2-compiler)
+The Tessel 2 compiler is another virtual machine with all the build tools needed to develope [native add-ons](https://nodejs.org/api/addons.html) to Node modules. These add-ons are built to perform faster or interact with lower level hardware than JS alone would provide. The compiler is being used, for example, to develop the [audio-video Node module](https://github.com/tessel/node-audiovideo) for webcams and recording audio from microphones.
+    
+### Device Related Code
+#####[Tessel 2 Operation System (OpenWRT)](https://www.github.com/tessel/openwrt-tessel)
+The primary processpr of the Tessel 2 runs a very lightweight version of Linux called OpenWRT. OpenWRT provides all of the TCP/IP drivers, threading/schedule support, and runs Node, Rust or whatever other language you're using with Tessel 2. This repo contains all of the source files of the Tessel build of OpenWRT.
+
+#####[Tessel 2 Co-processor Firmware](https://www.github.com/tessel/t2-firmware)
+Tessel 2 features a co-processor that manages a handful of responsibilities like routing USB communication from the CLI and the realtime operations on the GPIO pins of the two module ports. This repo contains not only the firmware that drives this microcontroller but hardware-related scripts that define the interface to other languages (like [tessel.js](https://www.github.com/tessel/t2-firmware/node/tessel.js) for Node scripts) so that the available hardware functionality doesn't get out of sync with what is exposed to the OpenWRT processor.
+
+### Hardware Designs
+#####[Tessel 2 Hardware Design Files](https://www.github.com/tessel/t2-hardware)
+These are the actual schematic and assembly files for the Tessel 2 hardware. You can find information about the parts that are used and how they are all connected to each other. These files were created with [KiCad](http://www.kicad-pcb.org/display/KICAD/KiCad+EDA+Software+Suite), a completely open source electronics design tool.
+#####[Tessel Parts Library](https://github.com/tessel/tm-kicad-library)
+This repo contains all of the KiCad part models for all Tessel hardware (not just Tessel 2).
```

Comment:
How do you use this? Do you need this in order to use the hardware design files? Or, these are the files that the schematics were generated from? Not sure what this repo is